### PR TITLE
Add Forge Metadata

### LINKED ISSUE #1787: Add forge build in env vars
```
CI_FORGE_TYPE = (gitea,github,...)
CI_FORGE_URL = github.com # gitea.com, codeberg.org, gitea.company.xyz ...
[CI_FORGE_API] = should we craft that too, or leave it to plugins?
```

did I missed something else that is usefull

might be usefull for plugins to address #1376

### PR DESCRIPTION
**Pull request by 6543**
close  #1787

### CODE REVIEW

File: server/forge/forge.go [hunk 1]
```diff
@@ -32,6 +32,9 @@ type Forge interface {
 	// Name returns the string name of this driver
 	Name() string
 
+	// Link returns the root url of a configured forge
+	Link() string
```
Comment (anbraten) [inline server/forge/forge.go:36]: Shouldn't we call it url then?

Comment (6543) [reply server/forge/forge.go:36]: no that is explizite link ...

https://woodpecker-ci.org/docs/next/usage/environment#built-in-environment-variables

Do use always use `_LINK` only one time we have URL ... and I'm for consistency ...

Comment (6543) [reply server/forge/forge.go:36]: -> #1793

Comment (6543) [reply server/forge/forge.go]: ok let's do a vote: https://framadate.org/mOJABOjBGDEVPsyv

**Review by anbraten — APPROVED** [review]

**Review by anbraten — APPROVED** [review]

**Pull request merged**